packages/strapi-design-system/src/Tag/Tag.js

```diff
@@ -25,35 +25,42 @@ const TagText = styled(Typography)`
 export const Tag = ({ children, icon, disabled, onClick, ...props }) => {
   const handleClick = (e) => {
     if (disabled) return;
-    onClick(e);
+    if (onClick) onClick(e);
   };
 
   return (
-    <TagWrapper
-      as="button"
+    <Box
+      display="inline-block"
       background={disabled ? 'neutral200' : 'primary100'}
       color={disabled ? 'neutral700' : 'primary600'}
       paddingLeft={3}
-      paddingRight={3}
-      onClick={handleClick}
-      aria-disabled={disabled}
       borderWidth="1px"
       borderStyle="solid"
       borderColor={disabled ? 'neutral300' : 'primary200'}
       hasRadius
       height={`${32 / 16}rem`}
       {...props}
     >
-      <Flex>
+      <Flex height="100%">
         <TagText disabled={disabled} variant="pi" fontWeight="bold" as="span">
           {children}
         </TagText>
 
-        <Box paddingLeft={2}>
-          <Flex>{icon}</Flex>
-        </Box>
+        <TagIconWrapper
```

Comment:
Hi @WalkingPizza, please make sure that this button have discernible text.

Since the parent element of this component is no longer a button, this new button needs a discernible text. For example, in the select component, we have an icon button to clear the value and this button has a little bit get a clearLabel attribute. It would be nice to have this kind of attribute in the Tag component as well (something like `actionLabel`). This way the user can pass a label depending on the action.